src/shared-module/files/component/__name@dasherize__.component.ts

```diff
@@ -1,8 +1,9 @@
-import { Component } from '@angular/core';
+import { Component<% if (isNgrxInstalled && hasNotRoot) { %>, ChangeDetectionStrategy<% } %> } from '@angular/core';
 
 @Component({
   selector: '<%= dasherize(((hasSection && !hasPage) ? (section + ' '): '') + ((hasSection && hasParentPage) ? (parentPage + ' ') : '') + ((hasSection && hasPage) ? (page + ' ') : '') + name) %>',
   templateUrl: '<%= dasherize(name) %>.html',
-  styleUrls: ['<%= dasherize(name) %>.scss']
+  styleUrls: ['<%= dasherize(name) %>.scss']<% if (isNgrxInstalled && hasNotRoot) { %>,
```

Comment:
@umbarutiba Could you please explain why we need to check `hasNotRoot` here? If I understood correctly, this check will prevent shared components from having `changeDetection` property in `@Component` decorator. Why shouldn't we add it for these components?

Reply:
@dmitryusenko   
![image](https://user-images.githubusercontent.com/49771379/75002149-a6931380-548d-11ea-9171-4e1a34d8907c.png)

Reply:
@umbarutiba @dmitryusenko let's discuss it. Maybe all components should have onPush strategy.

Reply:
@umbarutiba @dmitryusenko decided to leave as is